src/pluggy/_hooks.py

```diff
@@ -453,6 +451,53 @@ def _maybe_apply_history(self, method: "HookImpl") -> None:
                     result_callback(res[0])
 
 
+class _SubsetHookCaller(_HookCaller):
```

Comment:
in a  follow-up we might want to introduce a hook caller protocol as part of the exposed type cleanup

at a later point in time i would like to make pluggy able to work with a dependency topology of plugins 
(to better serparae conftests and eventually allow a dependency tree of plugins)

but thats for 2023 or later